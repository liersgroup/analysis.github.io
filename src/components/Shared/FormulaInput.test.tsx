import React, { useState } from 'react'
import { mount } from 'enzyme'

import Input, { DOWN, ENTER, RIGHT, UP } from './FormulaInput'

function getSuggestions() {
  return [
    'Alpha',
    'Bravo',
    'Charlie'
  ]
}

describe('FormulaInput', () => {
  function Component() {
    const [value, setValue] = useState("")

    return (
      <Input
        getSuggestions={getSuggestions}
        onChange={setValue}
        value={value}
      />
    )
  }

  it('suggests when the user starts typing', () => {
    const wrapper = mount(<Component />)

    // No suggestions are visible initially
    expect(wrapper.find('Suggestions').length).toEqual(0)
    expect(wrapper).toMatchSnapshot()

    // Suggestions appear as the user starts typing
    wrapper.
      find('input').
      simulate('change', { target: { value: 'a' } })

    expect(wrapper.find('Suggestions').length).toEqual(1)
    expect(wrapper).toMatchSnapshot()

    // The user can select suggestions with keypresses
    wrapper.
      find('input').
      simulate('keydown', { keyCode: DOWN }).
      simulate('keydown', { keyCode: RIGHT })

    expect(wrapper.find('input').prop('value')).toEqual('Alpha')
    expect(wrapper).toMatchSnapshot()

    // Subsequent updates only update the fragment
    wrapper.
      find('input').
      simulate('change', { target: { value: 'Alpha + C' } }).
      simulate('keydown', { keyCode: UP }).
      simulate('keydown', { keyCode: ENTER })

    expect(wrapper.find('input').prop('value')).toEqual('Alpha + Charlie')
    expect(wrapper).toMatchSnapshot()
  })
})
