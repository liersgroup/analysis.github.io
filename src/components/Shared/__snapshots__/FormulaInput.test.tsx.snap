// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormulaInput suggests when the user starts typing 1`] = `
<Component>
  <FormulaInput
    getSuggestions={[Function]}
    onChange={[Function]}
    value=""
  >
    <FormControl
      autoComplete="off"
      name="formula"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder=""
      value=""
    >
      <input
        autoComplete="off"
        className="form-control"
        name="formula"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        value=""
      />
    </FormControl>
  </FormulaInput>
</Component>
`;

exports[`FormulaInput suggests when the user starts typing 2`] = `
<Component>
  <FormulaInput
    getSuggestions={[Function]}
    onChange={[Function]}
    value="a"
  >
    <FormControl
      autoComplete="off"
      name="formula"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder=""
      value="a"
    >
      <input
        autoComplete="off"
        className="form-control"
        name="formula"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        value="a"
      />
    </FormControl>
    <Suggestions
      apply={[Function]}
      selected={null}
      suggestions={
        Array [
          "Alpha",
          "Bravo",
          "Charlie",
        ]
      }
    >
      <ListGroup>
        <ForwardRef
          as="div"
          className="list-group"
          onSelect={[Function]}
        >
          <div
            className="list-group"
            onKeyDown={[Function]}
          >
            <ListGroupItem
              action={true}
              active={false}
              disabled={false}
              key="Alpha"
              onClick={[Function]}
            >
              <ForwardRef
                as="button"
                className="list-group-item list-group-item-action"
                disabled={false}
                eventKey={null}
                onClick={[Function]}
              >
                <button
                  className="list-group-item list-group-item-action"
                  data-rb-event-key={null}
                  disabled={false}
                  onClick={[Function]}
                >
                  <Title
                    body="Alpha"
                  >
                    <span
                      key="h-1"
                    >
                      Alpha
                    </span>
                  </Title>
                </button>
              </ForwardRef>
            </ListGroupItem>
            <ListGroupItem
              action={true}
              active={false}
              disabled={false}
              key="Bravo"
              onClick={[Function]}
            >
              <ForwardRef
                as="button"
                className="list-group-item list-group-item-action"
                disabled={false}
                eventKey={null}
                onClick={[Function]}
              >
                <button
                  className="list-group-item list-group-item-action"
                  data-rb-event-key={null}
                  disabled={false}
                  onClick={[Function]}
                >
                  <Title
                    body="Bravo"
                  >
                    <span
                      key="h-1"
                    >
                      Bravo
                    </span>
                  </Title>
                </button>
              </ForwardRef>
            </ListGroupItem>
            <ListGroupItem
              action={true}
              active={false}
              disabled={false}
              key="Charlie"
              onClick={[Function]}
            >
              <ForwardRef
                as="button"
                className="list-group-item list-group-item-action"
                disabled={false}
                eventKey={null}
                onClick={[Function]}
              >
                <button
                  className="list-group-item list-group-item-action"
                  data-rb-event-key={null}
                  disabled={false}
                  onClick={[Function]}
                >
                  <Title
                    body="Charlie"
                  >
                    <span
                      key="h-1"
                    >
                      Charlie
                    </span>
                  </Title>
                </button>
              </ForwardRef>
            </ListGroupItem>
          </div>
        </ForwardRef>
      </ListGroup>
    </Suggestions>
  </FormulaInput>
</Component>
`;

exports[`FormulaInput suggests when the user starts typing 3`] = `
<Component>
  <FormulaInput
    getSuggestions={[Function]}
    onChange={[Function]}
    value="Alpha"
  >
    <FormControl
      autoComplete="off"
      name="formula"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder=""
      value="Alpha"
    >
      <input
        autoComplete="off"
        className="form-control"
        name="formula"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        value="Alpha"
      />
    </FormControl>
  </FormulaInput>
</Component>
`;

exports[`FormulaInput suggests when the user starts typing 4`] = `
<Component>
  <FormulaInput
    getSuggestions={[Function]}
    onChange={[Function]}
    value="Alpha + Charlie"
  >
    <FormControl
      autoComplete="off"
      name="formula"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder=""
      value="Alpha + Charlie"
    >
      <input
        autoComplete="off"
        className="form-control"
        name="formula"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        value="Alpha + Charlie"
      />
    </FormControl>
  </FormulaInput>
</Component>
`;
